h2o-algos/src/main/java/hex/schemas/GAMV3.java

```diff
@@ -0,0 +1,241 @@
+package hex.schemas;
+
+import hex.gam.GAM;
+import hex.gam.GAMModel;
+import hex.glm.GLMModel.GLMParameters;
+import hex.glm.GLMModel.GLMParameters.Solver;
+import water.api.API;
+import water.api.API.Direction;
+import water.api.API.Level;
+import water.api.schemas3.KeyV3.FrameKeyV3;
+import water.api.schemas3.ModelParametersSchemaV3;
+import water.api.schemas3.StringPairV3;
+
+public class GAMV3 extends ModelBuilderSchema<GAM, GAMV3, GAMV3.GAMParametersV3> {
+  public static final class GAMParametersV3 extends ModelParametersSchemaV3<GAMModel.GAMParameters, GAMParametersV3> {
+    public static final String[] fields = new String[] {
+            "model_id",
+            "training_frame",
+            "validation_frame",
+            "nfolds",
+            "seed",
+            "keep_cross_validation_models",
+            "keep_cross_validation_predictions",
+            "keep_cross_validation_fold_assignment",
+            "fold_assignment",
+            "fold_column",
+            "response_column",
+            "ignored_columns",
+            "ignore_const_cols",
+            "score_each_iteration",
+            "offset_column",
+            "weights_column",
+            "family",
+            "tweedie_variance_power",
+            "tweedie_link_power",
+            "theta", // equals to 1/r and should be > 0 and <=1, used by negative binomial
+            "solver",
+            "alpha",
+            "lambda",
+            "lambda_search",
+            "early_stopping",
+            "nlambdas",
+            "standardize",
+            "missing_values_handling",
+            "plug_values",
+            "compute_p_values",
+            "remove_collinear_columns",
+            "intercept",
+            "non_negative",
+            "max_iterations",
+            "objective_epsilon",
+            "beta_epsilon",
+            "gradient_epsilon",
+            "link",
+            "prior",
+            "lambda_min_ratio",
+            "beta_constraints",
+            "max_active_predictors",
+            "interactions",
+            "interaction_pairs",
+            "obj_reg",
+            "export_checkpoints_dir",
+            // dead unused args forced here by backwards compatibility, remove in V4
```

Comment:
Do we need these "dead/unused args forced here by backwards compatibility?" Seems there shouldn't be any backward compatibility issues with a new algo.

Reply:
@angela0xdata @Pscheidl 
Again, those parameters should stay.  Since GAM is essentially GLM with the column transformation.  This problem will go away once we get the inheritance in..